src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500RelativeDistinguishedName.cs

```diff
@@ -0,0 +1,153 @@
+            }
+        }
+
+        /// <summary>
+        ///   Gets a value that indicates whether this Relative Distinguished Name is composed
+        ///   of multiple attributes or only a single attribute.
+        /// </summary>
+        /// <value>
+        ///   <see langword="true"/> if the Relative Distinguished Name is composed of multiple
+        ///   attributes; <see langword="false"/> if it is composed of only a single attribute.
+        /// </value>
+        public bool HasMultipleElements => _singleElementType is null;
+
+        /// <summary>
+        ///   Gets the object identifier (OID) identifying the single attribute value for this
+        ///   Relative Distinguished Name (RDN), when the RDN only contains one attribute.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        ///   The Relative Distinguished Name has multiple attributes (<see cref="HasMultipleElements"/>
+        ///   is <see langword="true" />).
+        /// </exception>
+        public Oid GetSingleElementType()
+        {
+            if (_singleElementType is null)
+            {
+                throw new InvalidOperationException(SR.Cryptography_X500_MultiValued);
+            }
+
+            return _singleElementType;
+        }
+
+        /// <summary>
+        ///   Gets the textual representation of the value for the Relative Distinguished Name (RDN),
+        ///   when the RDN only contains one attribute.
+        /// </summary>
+        /// <returns>
+        ///   <see langword="null"/> if the encoded value is not a textual data type, such as an
+        ///   <c>OCTET STRING</c>; otherwise, the textual representation of the value.
+        /// </returns>
+        /// <exception cref="CryptographicException">
+        ///   The attribute is identified as a textual value, but the value did not successfully decode.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        ///   The Relative Distinguished Name has multiple attributes (<see cref="HasMultipleElements"/>
+        ///   is <see langword="true" />).
+        /// </exception>
+        public string? GetSingleElementValue()
+        {
+            if (_singleElementValue.IsEmpty)
+            {
+                throw new InvalidOperationException(SR.Cryptography_X500_MultiValued);
+            }
+
+            // X.520 defines a few non-textual attributes, such as objectIdentifier (2.5.4.106),
+            // which Windows renders textually as the bytes in hexadecimal preceded by an octothorpe,
+            // e.g. #06032A0304 for an objectIdentifier attribute whose value is the OID 1.2.3.4
+            //
+            // For these, we return null, and then let the X500Name.Format code handle the hex fallback.
+
+            try
+            {
+                AsnValueReader reader = new AsnValueReader(_singleElementValue.Span, AsnEncodingRules.BER);
```

Comment:
Just double checking that `BER` was intentional? Maybe a comment explaining why, and a test to solidify the behavior.

Reply:
[X500NameEncoder.ManagedDecode](https://github.com/dotnet/runtime/blob/6f13196cb096ef7c855fe1254214d25c578ad57e/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500NameEncoder.ManagedDecode.cs#L23) uses single DER reader so at very least this would be a deviation from that implementation.